apps/src/lib/node/ledger/shell/vote_extensions/val_set_update.rs

```diff
@@ -50,56 +55,32 @@ where
     pub fn validate_valset_upd_vext_and_get_it_back(
         &self,
         ext: validator_set_update::SignedVext,
-        last_height: BlockHeight,
+        signing_epoch: Epoch,
     ) -> std::result::Result<
         (token::Amount, validator_set_update::SignedVext),
         VoteExtensionError,
     > {
-        #[cfg(feature = "abcipp")]
-        if ext.data.block_height != last_height {
+        if self.storage.last_height.0 == 0 {
```

Comment:
Shouldn't we be checking the `epoch` inside the vext rather than our local node's `last_height`? Something like `if ext.data.epoch == Epoch(0)`

Reply:
it's fine for the epoch to be 0, what isn't fine is an extension that was signed at height 0. height 1 should be the first block height being decided

Reply:
if `last_height` has value 0, we are still deciding the first block, which couldn't have possibly had these protocol txs yet, because we only broadcast them at the commit phase of the first height